cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/LogAppender.java

```diff
@@ -61,13 +67,42 @@ public final void append(ILoggingEvent eventObject) {
       if (loggingContext == null) {
         return;
       }
+      addErrorCodeTags(eventObject);
       addExtraTags(eventObject, loggingContext);
     }
 
     LogMessage logMessage = new LogMessage(eventObject, loggingContext);
     appendEvent(logMessage);
   }
 
+  /**
+   * if event is for an exception deriving from ErrorCodeProvider, capture error group etc.
+   * @param event
+   */
+  private void addErrorCodeTags(ILoggingEvent event) {
+    if (event.getThrowableProxy() == null || !(event.getThrowableProxy() instanceof ThrowableProxy)) {
+      return;
+    }
+
+    Throwable throwable = ((ThrowableProxy) (event.getThrowableProxy())).getThrowable();
+    // add all error tags from all the inner exceptions
+    StringBuilder tagsSb = new StringBuilder();
+    while (throwable != null) {
+      if (throwable instanceof ErrorTagProvider) {
+
+        ErrorTagProvider errorCodeThrowable = (ErrorTagProvider) throwable;
+        for (ErrorTagProvider.ErrorTag tag: errorCodeThrowable.getErrorTags()) {
+          if (tagsSb.length() > 0) {
+            tagsSb.append(", ");
+          }
+          tagsSb.append(tag);
```

Comment:
it's possible to add duplicate tags if there are multiple in the stack trace. Would be better to keep track of them in a set to avoid dupes.

Reply:
Great point. I added a set.